virtool/indexes/api.py

```diff
@@ -31,89 +32,100 @@
+        Status Codes:
+            200: Successful operation
 
-    if not ready:
-        data = await virtool.indexes.db.find(db, req.query)
-        return json_response(data)
+        """
+        db = self.request.app["db"]
 
-    pipeline = [
-        {"$match": {"ready": True}},
-        {"$sort": {"version": -1}},
-        {
-            "$group": {
-                "_id": "$reference.id",
-                "index": {"$first": "$_id"},
-                "version": {"$first": "$version"},
-            }
-        },
-    ]
+        ready = self.request.query.get("ready", False)
 
-    ready_indexes = []
+        if not ready:
+            data = await virtool.indexes.db.find(db, self.request.query)
+            return json_response(data)
 
-    async for agg in db.indexes.aggregate(pipeline):
-        reference = await db.references.find_one(agg["_id"], ["data_type", "name"])
-
-        ready_indexes.append(
+        pipeline = [
+            {"$match": {"ready": True}},
+            {"$sort": {"version": -1}},
             {
-                "id": agg["index"],
-                "version": agg["version"],
-                "reference": {
-                    "id": agg["_id"],
-                    "name": reference["name"],
-                    "data_type": reference["data_type"],
-                },
-            }
-        )
+                "$group": {
+                    "_id": "$reference.id",
+                    "index": {"$first": "$_id"},
+                    "version": {"$first": "$version"},
+                }
+            },
+        ]
+
+        ready_indexes = []
+
+        async for agg in db.indexes.aggregate(pipeline):
+            reference = await db.references.find_one(agg["_id"], ["data_type", "name"])
+
+            ready_indexes.append(
+                {
+                    "id": agg["index"],
+                    "version": agg["version"],
+                    "reference": {
+                        "id": agg["_id"],
+                        "name": reference["name"],
+                        "data_type": reference["data_type"],
+                    },
+                }
+            )
 
-    return json_response(ready_indexes)
+        return json_response(ready_indexes)
 
 
-@routes.get("/indexes/{index_id}")
+@routes.view("/indexes/{index_id}")
 @routes.jobs_api.get("/indexes/{index_id}")
-async def get(req):
-    """
-    Get the complete document for a given index.
+class IndexView(PydanticView):
+    async def get(self) -> Union[r200[GetIndexResponse], r404]:
+        """
+        Get the complete document for a given index.
 
-    """
-    db = req.app["db"]
-    pg = req.app["pg"]
+        Status Codes:
+            200: Successful operation
+            404: Not found
 
-    index_id = req.match_info["index_id"]
+        """
+        db = self.request.app["db"]
+        pg = self.request.app["pg"]
 
-    document = await db.indexes.find_one(index_id)
+        index_id = self.request.match_info["index_id"]
```

Comment:
Use `aiohttp-pydantic` support for path parameter validation.

https://github.com/Maillol/aiohttp-pydantic#inject-path-parameters